Refactoring code to deduplicate common logic

ChrisCarini opened the pull request:
Fixes #1

ChrisCarini committed:
Refactoring code to deduplicate common logic

ChrisCarini commented:
@alonelysaber - can you take a look at this PR when you have a chance?

alonelysaber commented:
Hey Chris!
I'm so sorry I've left you guys hanging for such a long time, I've been
buried by school work and life in general. For the past 3+ months I've
barely coded anything at all. I have been seeing your emails though and I
really appreciate the work you and ejach have put into my what I consider a
dogshit project (seriously it's shamefully bad). It's also pretty shameful
to admit that I have very little knowledge of how Github works and will be
getting around to learning and relearning the fundamentals again so I can
get back into the groove of things and make more projects soon. For now, if
I understand the timeline correctly, ejach first opened an issue(?) in my
repository and fixed some things before closing the issue.

After that, you opened a pull request (again, not entirely sure what that
means) and fixed some things along with ejach.

I have taken a look at the code again and understand very little about it
(please forgive my stupidity but github's UI is still very confusing to me).
For now I would love to accept whatever pull request you want me to approve
and merge(?) your fixes into my code and that should hopefully make things
easier to work with.

Thanks again, looking forward to collaborating with you guys soon.

On Fri, Mar 25, 2022 at 9:28 AM Chris Carini ***@***.***>
wrote:

> @alonelysaber <https://github.com/alonelysaber> - can you take a look at
> this PR when you have a chance?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alonelysaber/Fixture-Calendar/pull/3#issuecomment-1078628777>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVRWETTD4BE55OCJPQ7QYFLVBU2V5ANCNFSM5PPP5VSQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ChrisCarini commented:
> Hey Chris! I'm so sorry I've left you guys hanging for such a long time, I've been buried by school work and life in general.

No problem - such is life! :)

> I have been seeing your emails though and I really appreciate the work you and ejach have put into my what I consider a dogshit project (seriously it's shamefully bad).

Don't be so harsh on yourself! Remember, we all started somewhere

> It's also pretty shameful to admit that I have very little knowledge of how Github works and will be getting around to learning and relearning the fundamentals again so I can get back into the groove of things and make more projects soon.

Not shameful at all - you should celebrate not knowing things and taking interest in learning them. No shame.

> For now, if I understand the timeline correctly, ejach first opened an issue(?) in my repository and fixed some things before closing the issue. After that, you opened a pull request (again, not entirely sure what that means) and fixed some things along with ejach.

Yup, sounds about right :)

> ...opened a pull request (again, not entirely sure what that means)

A "pull request" is simply someone proposing a change to a repo / code. [GitHub has some pretty good docs "About pull requests"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests).

> I have taken a look at the code again and understand very little about it (please forgive my stupidity but github's UI is still very confusing to me). 

No problem - feel free to take your time and get used to the UI and take a look at the code. There are actually two different ways to 'view' a diff - "split view" and "unified". People have different preferences, so one might be easier for you to digest. for me personally, I prefer 'split view'. [Here is a link to the docs that shows how to change between the views.](https://docs.github.com/es/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests#diff-view-options)

> For now I would love to accept whatever pull request you want me to approve and merge(?) your fixes into my code and that should hopefully make things easier to work with.

Whatever you prefer! I think it would still be a good learning exercise for @ejach (if they are interested, of course! No pressure!) to reopen #1 and shepherd their original change through. Ultimately, though, it is up to you, @alonelysaber to decide what you would prefer and like to do, since after all, you are the owner of this repo on GitHub! :)

> Thanks again, looking forward to collaborating with you guys soon.

No problem, and same! :)
